shared/plan/service.py

```diff
@@ -46,7 +47,14 @@ def __init__(self, current_org: Owner):
         Raises:
             ValueError: If the organization's plan is unsupported.
         """
-        self.current_org = current_org
+        if (
+            current_org.service == Service.GITLAB.value
+            and current_org.parent_service_id
```

Comment:
This doesn't seem correct... you are checking for `current_org.parent_service_id` not being `None`, but `self.current_org` is getting the value of `current_org.root_organization`